aptos/contracts/sources/contract_upgrade.move

```diff
@@ -5,20 +5,25 @@ module wormhole::contract_upgrade {
     use wormhole::deserialize;
     use wormhole::cursor;
     use wormhole::vaa;
-    use wormhole::wormhole;
+    use wormhole::state;
 
     const E_UPGRADE_UNAUTHORIZED: u64 = 0;
     const E_UNEXPECTED_HASH: u64 = 1;
     const E_INVALID_MODULE: u64 = 2;
     const E_INVALID_ACTION: u64 = 3;
+    const E_INVALID_TARGET: u64 = 4;
 
     // TODO(csongor): document how this works
     struct UpgradeAuthorized has key {
         hash: vector<u8>
     }
 
+    struct Hash {
```

Comment:
What is the rationale behind passing around a Hash {vector<u8>} instead of simply a vector<u8>?

Reply:
this is personal preference, I generally like to create wrapper types for things that are semantically well-defined, like a "hash". IMO it makes the code more readable because it's more clear what parse payload returns now - it's a little odd to me if the parser takes unstructured data (vector<u8>) and returns the same type of unstructured data - as if it didn't do anything!
It's also easier to see that authorize_upgrade takes the output of parse_payload just by looking at the types